### PR TITLE
Simple progress bar

### PR DESCRIPTION
**Pull request by marco-cardoso**
#244  This version of the progress bar should work in a normal CLI and also in jupyter notebook.  I tried to figure out a way to generate a more realistic update status, but for the sake of simplicity I used fixed percentage values. If you know about a better way to do that, please let me know.

### CODE REVIEW

File: pandas_profiling/utils/notebook.py [hunk 1]
```diff
@@ -6,3 +6,13 @@ def full_width():
     from IPython.core.display import display, HTML
 
     display(HTML("<style>.container { width:100% !important; }</style>"))
+
+
+def in_ipynb() -> bool:
```
Comment (marco-cardoso) [inline pandas_profiling/utils/notebook.py]: I've not found a specific method to do this in the another modules, so I decided to add it. Let me know if there's a better approach to detect the environment.

Comment (sbrugman) [reply pandas_profiling/utils/notebook.py]: > It is also possible to let tqdm automatically choose between console or notebook versions by using the autonotebook submodule:
> ```from tqdm.autonotebook import tqdm```

https://github.com/tqdm/tqdm

**Comment by sbrugman** [comment]
Great work, @marco-cardoso. 

Some points to complete the contribution:
-  `tdqm` itself contains a mechanism to differntiate between environments (see above). 
- It should be possible to disable the progressbar (`tqmd(disable=...)`).
- There are failing tests (`ImportError: IntProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html`).
- Please run the code linting and regenerate the docs (`make lint` and `make docs`, more [details](https://github.com/pandas-profiling/pandas-profiling/blob/master/Makefile)).

**Comment by marco-cardoso** [comment]
Thanks for the points @sbrugman . I did a couple of changes and now the codes are working and the documents are updated. The default option is not to show the progress bar, in order to activate it is necessary to set the parameter progress_bar= True when calling the method profile_report.

**Comment by appleyuchi** [comment]
But I did not see any change in my kaggle notebook
Which edition of profiling is progress bar  supported with?

**Comment by appleyuchi** [comment]
Config parameter "progress_bar" does not exist.

**Comment by marco-cardoso** [comment]
Hello @appleyuchi , thanks for your feedback.

Could you tell me which version did you use?

I tested the version from my account, available [here](https://github.com/marco-cardoso/pandas-profiling/tree/progress-bar) (since this PR was not approved yet), and after executing 'python setup.py install' and importing pandas_profiling in a script, an error was raised because TQDM was not installed. After I fixed it, by installing with pip separated, the progress bar worked as expected. This made me realize that I forgot to update the setup.py with the progress bar dependencies (Only updated the requirements.txt). 

Also, I recorded a short video with the progress bar working with a small dataset.

ezgif-2-e392bd305f71

**Comment by marco-cardoso** [comment]
I'm trying to update this, but something is wrong. Do I need to write unit tests for the methods which I created in order for the PR to pass the coverage tests?

**Comment by appleyuchi** [comment]
> Hello @appleyuchi , thanks for your feedback.
> 
> Could you tell me which version did you use?
> 
> I tested the version from my account, available [here](https://github.com/marco-cardoso/pandas-profiling/tree/progress-bar) (since this PR was not approved yet), and after executing 'python setup.py install' and importing pandas_profiling in a script, an error was raised because TQDM was not installed. After I fixed it, by installing with pip separated, the progress bar worked as expected. This made me realize that I forgot to update the setup.py with the progress bar dependencies (I had only updated the requirements.txt).
> 
> Also, I recorded a short video with the progress bar working with a small dataset.
> 
> ezgif-2-e392bd305f71

Thanks for your replies~!
My Environment is as follows:
Python version:3.6.7rc1
tqdm                   4.34.0
pandas                 0.23.4   
pandas-profiling       2.3.0

**Comment by sbrugman** [comment]
@marco-cardoso Many thanks for kickstarting this feature. A lot has changed since the PR, so it seemed to me to be more effective to implement it in a new PR than resolve the merge conflicts (#345).

**Pull request closed without merging**